### DISCUSSION TITLE
RX-A2A works!

### DISCUSSION THREAD

**Discussion by orkestrark** [mvdwetering/yamaha_ynca#282] [General]
Thank you! It actually works great.

i wonder if it would be possible to have an entity with the number of channels from the current source. That would be a way to automatically switch sound mode with an automation.

a track is in 5.1 or more? Switch to Straight.
a track is stereo ? Switch to All-Ch Stereo

**Comment by mvdwetering** [comment]
Thanks for the confirmation on the RX-A2A.

I don't think there is a command to see that, but we can check to make sure.

To check I need integration diagnostics after you play some content. 
So play some 5.1 content, then stereo content and then download diagnostics and _attach_ the file here. 

Just to be clear I mean the diagnostics you get from the vertical ... menu on the integration
![image](https://github.com/mvdwetering/yamaha_ynca/assets/732514/6e014018-d42d-40fb-8948-940a82dd081a)

**Comment by orkestrark** [comment]
[config_entry-yamaha_ynca-d8ea7b79bec8a700b9c010daa190cca0 - 51.json](https://github.com/mvdwetering/yamaha_ynca/files/15197573/config_entry-yamaha_ynca-d8ea7b79bec8a700b9c010daa190cca0.-.51.json)
[config_entry-yamaha_ynca-d8ea7b79bec8a700b9c010daa190cca0 - stereo.json](https://github.com/mvdwetering/yamaha_ynca/files/15197574/config_entry-yamaha_ynca-d8ea7b79bec8a700b9c010daa190cca0.-.stereo.json)

There you go. Thank you so much for your work!

**Reply by mvdwetering** [reply]
Unfortunately there seems nothing in there that indicates amount of channels or type of content, so nothing I can do.

As an alternative maybe the source you use can indicate the amount of channels or type of content if it integrates with Home Assistant?

**Reply by orkestrark** [reply]
I checked the files before submitting them and indeed I didn’t see anything promising. My sources don’t send audio channel info unfortunately.

**Comment by orkestrark** [comment]
Hi again mvd! I found a workaround to get audio info in HA via Scrape. Audio info is actually available here: 

IP_YAMAHA/YamahaExtendedControl/v1/main/getSignalInfo

Then an automation and a sensor do the rest of the job.

I don't know if that's something you can somehow integrate into YNCA, anyway I wanted to share the info with you.

**Reply by mvdwetering** [reply]
I want to keep this integration limited to the YNCA protocol, so adding that would be out so scope for this integration.

However that URL seems to be part of MusicCast protocol (only based on a very quick Google search) 

You could try the built in [Home Assistant MusicCast integration](https://www.home-assistant.io/integrations/yamaha_musiccast/) to see if the info you want is in there. It does not seem to be listed in the documentation, but maybe it is out of date.

If it is not in there you can submit a feature request here: https://github.com/orgs/home-assistant/discussions
